src/coreclr/runtime/portable/AllocFast.cpp

```diff
@@ -5,11 +5,18 @@
 #include <fcall.h>
 
 extern void RhExceptionHandling_FailedAllocation(MethodTable *pMT, bool isOverflow);
+EXTERN_C Object* RhpGcAlloc(MethodTable* pMT, uint32_t uFlags, uintptr_t numElements, void * pTransitionFrame);
 
-EXTERN_C FCDECL2(Object*, RhpNewVariableSizeObject, MethodTable* pMT, INT_PTR size)
+EXTERN_C FCDECL2(Object*, RhpNewVariableSizeObject, MethodTable* pMT, INT_PTR numElements)
 {
-    PORTABILITY_ASSERT("RhpNewVariableSizeObject is not yet implemented");
-    return nullptr;
+    Object* obj = RhpGcAlloc(pMT, 0, numElements, nullptr);
+    if (obj == NULL)
+    {
+        RhExceptionHandling_FailedAllocation(pMT, false /* isOverflow */);
+        return nullptr;
```

Comment:
```suggestion
```
You can delete redundant `return NULL` in the existing allocator in this file as well.